src/test/1.3/scala/com/holdenkarau/spark/testing/SampleScalaCheckTest.scala

```diff
@@ -69,6 +69,66 @@ class SampleScalaCheckTest extends FunSuite
     check(property)
   }
 
+  test("test RDD sized generator (nodes)") {
+    val nodeGenerator = RDDGenerator.genSizedRDD[MyNode](sc) { size: Int =>
+      def genGraph(sz: Int): Gen[MyNode] = for {
+          id <- Arbitrary.arbitrary[Int]
+          size <- if (sz <= 0) Gen.const(0) else Gen.choose(0, sz)
+          nodes <- Gen.listOfN(size, genGraph(sz / 2))
+        } yield {
+        MyNode(id, nodes)
+      }
+      genGraph(size)
+    }
+
+    val property =
+    forAll(nodeGenerator) {
+      rdd =>
+        rdd.map(_.key).count == rdd.count
+    }
+    check(property)
+  }
+
+
+  test("test custom RDD sized generator") {
+
+    val humanGen: Gen[RDD[List[Human]]] =
+      RDDGenerator.genSizedRDD[List[Human]](sc) { size: Int =>
+
+        val adultsNumber = math.ceil(size * 0.8).toInt
+        val kidsNumber = size - adultsNumber
+
+        def genHuman(genAge: Gen[Int]): Gen[Human] = for {
+            name <- Arbitrary.arbitrary[String]
+            age <- genAge
+        } yield Human(name, age)
+
+        for {
+            kids <- Gen.listOfN(kidsNumber, genHuman(Gen.choose(0, 18)))
+            adults <- Gen.listOfN(adultsNumber, genHuman(Gen.choose(18, 130)))
+        } yield {
+          kids ++ adults
+        }
+      }
+
+    val property =
+    forAll(humanGen.map(_.flatMap(identity))) {
+      rdd =>
```

Comment:
Since this is a sized generator does it make sense to look at the size chosen and the size of the RDD? It's not a test I would expect a user to do, but it would be a good test for the library its self.